lldb/source/Core/ValueObject.cpp

```diff
@@ -615,7 +615,15 @@ bool ValueObject::GetSummaryAsCString(TypeSummaryImpl *summary_ptr,
       m_synthetic_value->UpdateValueIfNeeded(); // the summary might depend on
                                                 // the synthetic children being
                                                 // up-to-date (e.g. ${svar%#})
-    summary_ptr->FormatObject(this, destination, actual_options);
+
+    TargetSP target = GetExecutionContextRef().GetTargetSP();
+    if (target) {
+      /// Construct RAII types to time and collect data on summary creation.
+      SummaryStatistics::SummaryInvocation summary_inv =
+          target->GetSummaryStatisticsForProvider(*summary_ptr);
```

Comment:
I think this wouldn't work anymore because you're returning a copy of the `SummaryInvocation` right? My previous suggestion was to have `SummaryInvocartion` update the `SummaryStatisticsCache` when it's done.

Reply:
This actually does (humorously in my opinion) work

I didn't realize the intent was for the RAII object to update the cache, I actually agree that is better and frees us from needing std::map for any reference invalidation. I will modify.